features/jest/client/src/main/java/org/opennms/plugins/elasticsearch/rest/template/DefaultTemplateLoader.java

```diff
@@ -48,8 +54,9 @@ public String load(Version serverVersion, String resource) throws IOException {
         for (int i = serverVersion.getMajor(); i >= 0; i--) {
             final String versionSuffix = i == 0 ? "" : String.format(".es%d", i);
             final String resourceWithSuffix = String.format("%s%s.json", resource, versionSuffix);
-            LOG.debug("Attempting to find template with resource name: {} (requested: {})", resourceWithSuffix, resource);
-            final String template = getResource(resourceWithSuffix);
+            LOG.debug("Attempting to find template with resource name: {} (requested: {})",
+                      resourceWithSuffix, resource);
+            final String template = getTemplate(resourceWithSuffix);
```

Comment:
What's the purpose of changing this logic?

I suspect it may be for dealing with an update pattern string in the template. If this is the case, we should be updating the template pattern within the existing documents as opposed to forcing the user to provide a modified one themselves.

Reply:
Actually over here we are not implying that the user should provide indexPrefix value, it can be opted through property file, Check is made for template in opennms home directory path if not found source code document will be picked.

private String getModifiedIndex() {
        if (!eventIndexName.isEmpty()) {
            return String.format("%s-%s", eventIndexName, INDEX_PREFIX);
        }
        return INDEX_PREFIX;
  }
Above piece of code provides the dynamic creation without affecting existing functionality.